src/plugins/home/public/application/components/home.tsx

```diff
@@ -22,16 +24,34 @@ import { Welcome } from './welcome';
 
 const KEY_ENABLE_WELCOME = 'home:welcome:show';
 
-export class Home extends Component {
-  constructor(props) {
+export interface HomeProps {
+  addBasePath: (url: string) => string;
+  directories: FeatureCatalogueEntry[];
+  solutions: FeatureCatalogueSolution[];
+  localStorage: Storage;
+  urlBasePath: string;
+  telemetry: TelemetryPluginStart;
+  indexPatternService: IndexPatternsContract;
+}
+
+interface State {
+  isLoading: boolean;
+  isNewKibanaInstance: boolean;
+  isWelcomeEnabled: boolean;
+}
+
+export class Home extends Component<HomeProps, State> {
+  private _isMounted?: boolean;
```

Comment:
nit: `_isMounted = false`. Why it should have a third state with `undefined`?

Reply:
I didn't want to change the current behaviour. But you're right. The `undefined` state is not explicitly used.